v3/src/components/graph/models/data-configuration-model.ts

```diff
@@ -197,20 +221,36 @@ export const DataConfigurationModel = types
     get uniqueAttributes() {
       return Array.from(new Set<string>(this.attributes))
     },
-    get tipAttributes() {
+    get tipAttributes():RoleAttrIDPair[] {
       return TipAttrRoles
-        .map(place => self.attributeID(place))
-        .filter(id => !!id) as string[]
+        .map(role => {
+          return {role, attributeID: self.attributeID(role) }
+        })
+        .filter(pair => !!pair.attributeID)
     },
     get uniqueTipAttributes() {
-      return Array.from(new Set<string>(this.tipAttributes))
+      const tipAttributes = this.tipAttributes,
+        idCounts: {[id: string]: number} = {}
+      tipAttributes.forEach((aPair:RoleAttrIDPair) => {
+        idCounts[aPair.attributeID] = (idCounts[aPair.attributeID] || 0) + 1
+      })
+      return tipAttributes.filter((aPair:RoleAttrIDPair) => {
+        if (idCounts[aPair.attributeID] > 1) {
+          idCounts[aPair.attributeID]--
+          return false
+        }
+        return true
+      })
     },
     get noAttributesAssigned() {
       // The first attribute is always assigned as 'caption'. So it's really no attributes assigned except for that
       return this.attributes.length <= 1
     },
     get numberOfPlots() {
-      return self._yAttributeDescriptions.length
+      return this.filteredCases.length
```

Comment:
This warrants a comment -- I initially read it as implying that there's a plot for each case.

Reply:
Added comment